Sources/MapboxDirections/Directions.swift

```diff
@@ -557,6 +558,7 @@ open class Directions: NSObject {
      - returns: A GET or POST HTTP request to calculate the specified options.
      */
     open func urlRequest(forCalculating options: DirectionsOptions) -> URLRequest {
+        if options.waypoints.count < 2 { assertionFailure("waypoints array requires at least 2 waypoints") }
```

Comment:
Moving this assertion to these methods doesn't seem to make any difference. We will still crash if `options.waypoints.count < 2` in production.

What if we return an invalid URL here? Or an URL of Directions API but with `{coordinates}` as is including something like `[]` or `empty` if the waypoints array is empty.

Reply:
I think what this does is prevents the crash from happening before we fetch a response. We could provide an invalidInput instead of an assertion failure, but again we'd see this crash in production. Providing an invalid URL could be a good idea. If we provide a URL of the Directions API with empty endpoints, I think we'd see a similar crash.

Reply:
My point here is that SDK should not crash on incorrect input. We have a way to return an error when requesting routes and should not crash there too.

All these cases should be fixed and covered with tests (we can mock a response that happens if you provide 1 or 0 waypoints).